include/unifex/get_stop_token.hpp

```diff
@@ -16,15 +16,44 @@
 #pragma once
 
 #include <unifex/stop_token_concepts.hpp>
+#include <unifex/inplace_stop_token.hpp>
 #include <unifex/tag_invoke.hpp>
 #include <unifex/type_traits.hpp>
+#include <unifex/coroutine.hpp>
 #include <unifex/unstoppable_token.hpp>
 
+#if !UNIFEX_NO_COROUTINES
+#include <unifex/await_transform.hpp>
+#endif
+
 #include <unifex/detail/prologue.hpp>
 
 namespace unifex {
 namespace _get_stop_token {
   inline const struct _fn {
+  private:
+#if !UNIFEX_NO_COROUTINES
+    template <typename StopToken>
+    struct awaiter_ {
+      StopToken stoken_;
+      bool await_ready() const noexcept {
+        return true;
+      }
+      void await_suspend(coro::coroutine_handle<>) const noexcept {
+      }
+      StopToken await_resume() noexcept {
+        return (StopToken&&) stoken_;
+      }
+    };
+    template <typename StopToken>
+    awaiter_(StopToken) -> awaiter_<StopToken>;
+
+    template <typename Promise>
+    friend auto tag_invoke(tag_t<await_transform>, Promise& promise, _fn get_stop_token) noexcept {
+      return awaiter_{get_stop_token(promise)};
+    }
```

Comment:
I feel like you could do this more generically in the implementation of `await_transform`.  Every awaitable _Receiver_ query CPO is going to have the same shape.  Couldn't you add a case to [the default implementation](https://github.com/facebookexperimental/libunifex/blob/master/include/unifex/await_transform.hpp#L207) that looks like
```
else if constexpr (is_receiver_query_cpo_v<Value> && is_invocable<Value, Promise>) {
  return awaiter_{value(promise)};
}
```

You might need to tweak the `awaiter_` type a little if there's such a thing as a _Receiver_ query CPO that returns something immovable and uncopyable (maybe have `await_resume` return a reference to the value?), but it ought to otherwise work for every promise/CPO pair that the given promise supports.

Reply:
If I go with Lewis' suggestion and make it `co_await get_stop_token()`, then `Value` is whatever `get_stop_token()` returns. I would need to make that also a receiver query CPO for your code to work, and it would need to behave exactly like `get_stop_token`. Which makes me tempted to just make `get_stop_token()` return `get_stop_token`.

Reply:
> You might need to tweak the awaiter_ type a little if there's such a thing as a Receiver query CPO that returns something immovable and uncopyable

I don't _think_ we have a receiver query that returns an immovable thing by value.